club/models.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+
+class Member(AbstractUser):
+    email = models.EmailField(unique=True)
+    type = models.CharField(max_length=20)
+    address = models.CharField(max_length=100)
+    home_phone = models.CharField(max_length=20)
+    mobile = models.CharField(max_length=20)
+    date_added = models.DateTimeField(auto_now_add=True)
+    
+    def __str__(self):
+        return f'{self.first_name} {self.last_name}'
+    
+    
+class Booking(models.Model):
+    IS_RECURRING_CHOICES = [(False, 'No'), 
+                            (True, 'Yes')]
+    time = models.DateTimeField()
+    start_date = models.DateTimeField()    
+    end_date = models.DateTimeField()
+    date_added = models.DateTimeField(auto_now_add=True)
+    recurring = models.BooleanField(default=False, 
+                                    choices=IS_RECURRING_CHOICES)
+    member = models.ForeignKey(Member, 
+                               on_delete=models.SET_NULL, 
+                               related_name='booking')
+
+    class Meta:
+        ordering = ['-time']
+
+    def __str__(self):
+        return f'{self.time} - {self.member}'
+    
+
+class Guests(models.Model):
+    players = models.ManyToManyField(Booking)
+    date_added = models.DateTimeField(auto_now_add=True)
+    member_id = models.CharField(max_length=10)
+
+    def __str__(self):
+        return f'{self.players}'
+
+
+class Subscription(models.Model):
+    type = models.CharField(max_length=20)
+    amount = models.PositiveIntegerField()
+    member = models.ForeignKey(Member, 
+                               on_deletee=models.PROTECT, 
+                               related_name='subscription')
+    date = models.DateField()
```

Comment:
this is start / renewal date right?

Reply:
It really should be for the year. The subscriptions work by year, regardless of when you pay. If you pay in january or october 2023. It is for the year 2023

Reply:
ok cool, thanks for clarifying, in that case can we just use an int?

<img width="718" alt="image" src="https://github.com/theonlysam/bookthecourt/assets/387927/c566aa7b-d406-4c4e-b7e8-793568ef9e2c">